utils/treesitter/testdata

```diff
@@ -0,0 +1 @@
+../../toolchain/parser/testdata
```

Comment:
Please don't include tests by symlink. Any tests should probably instead be exposed as a target instead, to make dependencies clearer.

However, the toolchain's parser is also not a great baseline to compare with -- we're going to make pragmatic choices about whether diagnostics belong in the lexer, parser, or semantics. Also, as the support expands and more invalid syntax use-cases are handled, it should be expected that the parser will skew further from tree-sitter behavior. If you're looking for better tests to compare with, the explorer might be a better choice because the passing tests there are all going to be semantically valid.